fhir-server/src/main/java/au/csiro/pathling/fhirpath/literal/DateLiteralPath.java

```diff
@@ -53,28 +56,38 @@ public static DateLiteralPath fromString(@Nonnull final String fhirPath,
     check(context.getIdColumn().isPresent());
     final String dateString = fhirPath.replaceFirst("^@", "");
     java.util.Date date;
+    DateLiteralFormat format;
     // Try parsing out the date using the three possible formats, from full (most common) down to
     // the year only format.
     try {
       date = DatePath.getFullDateFormat().parse(dateString);
```

Comment:
SimpleDataFormat is not thread safe. Since we use a singleton here (per spark executor) we might run into synchronisation issues with muti-threaded executors. 
I think the solution maybe to use thread local rather than static variable.

Reply:
See 8f36d1434838ddced3e5e67c4e57d6bb77e3b7a5.